### PR TITLE
Editing SIT requests as TSP

### PR DESCRIPTION
**Pull request by LeDeep**
## Description

This PR sets the fields that a TSP can edit in SIT request, based on the state of the SIT request.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make client_run
```

For TSP SIT request status `REQUESTED`
- All fields will be editable

For TSP SIT request status `IN_SIT`
- `Actual start date` && `notes` fields are editable
- User cannot edit the actual start date to a day that is before the earliest authorized start date
- all other fields are read only

For TSP SIT request status `RELEASED`
- `Date out` and `notes` fields are editable
- all other fields are read only

For TSP SIT request status `DELIVERED`
- `Start date` `Date out` && `notes` fields are editable
- User can edit the start date to a date between the earliest authorized start date and the date out
- User can edit the date out to day on or after the current date out
- all other fields are read only

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/166049008)
* [Pivotal story](https://www.pivotaltracker.com/story/show/165900387)
* [Pivotal story](https://www.pivotaltracker.com/story/show/166049362)
* [Pivotal story](https://www.pivotaltracker.com/story/show/166049743)

### CODE REVIEW

**Comment by jacquelineIO** [comment]
Looks good, needs one change, I think the Original Released Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049362 should be identical to the Destination Delivered Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049743

`Actual start date` should be editable in both. It was updated and fixed in Delivered story but wasn't fixed in Released story.

File: cypress/integration/tsp/storageInTransit.js [hunk 1]
```diff
@@ -521,3 +530,107 @@ function tspUserDeletesSitRequest() {
...
+
+  cy.location().should(loc => {
+    expect(loc.pathname).to.match(/^\/shipments\/[^/]+/);
+  });
+
+  cy
+    .get('[data-cy=storage-in-transit-panel] [data-cy=sit-edit-link]')
+    .contains('Edit')
+    .click()
+    .get('input[name=actual_start_date]')
```
[the remark this answers is not in the document]
Comment (LeDeep) [reply cypress/integration/tsp/storageInTransit.js]: might make it easier to read, not sure if there is also a performance gain. Will look into and refactor as needed.

[the remark this answers is not in the document]
Comment (LeDeep) [reply cypress/integration/tsp/storageInTransit.js]: > Looks good, needs one change, I think the Original Released Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049362 should be identical to the Destination Delivered Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049743
> 
> `Actual start date` should be editable in both. It was updated and fixed in Delivered story but wasn't fixed in Released story.

This is fixed.

**Comment by LeDeep** [comment]
> Looks good, needs one change, I think the Original Released Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049362 should be identical to the Destination Delivered Edit https://www.pivotaltracker.com/n/projects/2136865/stories/166049743
> 
> `Actual start date` should be editable in both. It was updated and fixed in Delivered story but wasn't fixed in Released story.

This is fixed.

File: cypress/integration/tsp/storageInTransit.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeDeep) [reply cypress/integration/tsp/storageInTransit.js]: Looked over the cypress docs and updated the tests to read `cy.get` as that is the preferred method

File: src/shared/StorageInTransit/StorageInTransitTspEditForm.jsx [hunk 2]
```diff
@@ -34,13 +34,7 @@ export class StorageInTransitTspEditForm extends Component {
...
     const isReleased = storageInTransit.status === 'RELEASED';
     const isDelivered = storageInTransit.status === 'DELIVERED';
-    const outDateMinDate = new Date(storageInTransit.out_date);
-    const utcOutDate = new Date(
-      outDateMinDate.getUTCFullYear(),
-      outDateMinDate.getUTCMonth(),
-      outDateMinDate.getUTCDate(),
-    );
-    const disabledDaysForOutDayPicker = [{ before: utcOutDate }];
+    const disabledDaysForOutDayPicker = [{ before: utcMinDate }];
```
Comment (jacquelineIO) [inline src/shared/StorageInTransit/StorageInTransitTspEditForm.jsx]: I think `utcMinDate` is the earliest authorized start date `minDate={this.props.storageInTransit.authorized_start_date}`. You need the `minOutDate=Actual Start Date`

Comment (LeDeep) [reply src/shared/StorageInTransit/StorageInTransitTspEditForm.jsx]: you're right. I was mixing up `actual_start_date` and `authorized_start_date`

**Pull request merged**